warp10/src/main/java/io/warp10/continuum/ingress/Ingress.java

```diff
@@ -1772,7 +1807,10 @@ public void close() throws Exception {
                   continue;
                 }
               }
-              pushDeleteMessage(start, end, minage, metadata);
+              
+              if (!metaonly) {
+                pushDeleteMessage(start, end, minage, metadata);
+              }
```

Comment:
It does not change much as the pushDeleteMessages in the finally clause simply flushes unsent messages, but I added a conditional. For completeDeletion, it will be set to true when metaonly is specified so the code within the conditional will be called.